pkg/asset/installconfig/alibabacloud/validation.go

```diff
@@ -100,6 +121,92 @@ func validateResourceGroup(client *Client, ic *types.InstallConfig, path *field.
 	return append(allErrs, field.NotFound(path.Child("resourceGroupID"), ic.AlibabaCloud.ResourceGroupID))
 }
 
+func validateVpc(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+	vpcs, err := client.ListVpcs(ic.AlibabaCloud.VpcID)
+	if err != nil {
+		return append(allErrs, field.InternalError(path.Child("vpcID"), err))
+	}
+	if vpcs.TotalCount == 0 {
+		allErrs = append(allErrs, field.NotFound(path.Child("vpcID"), ic.AlibabaCloud.VpcID))
+	}
+	return allErrs
+}
+
+func validateVSwitches(client *Client, ic *types.InstallConfig, path *field.Path) field.ErrorList {
+	allErrs := field.ErrorList{}
+
+	zoneIDs := []string{}
```

Comment:
At the end, there should be validation that there is a vSwitch for every zone specified in a machine pool. For example, if the control-plane machine pool specifies that zone Z should be used for control plane nodes, then there *must* be a vSwitch specified in zone Z.

Reply:
Yes, I think it is more appropriate to add  the validation of the machine pool, I will create a new PR.